library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java

```diff
@@ -0,0 +1,168 @@
+     */
+    @Override
+    protected void starting(Description description)
+    {
+      super.starting(description);
+      outputPath = new File("target/" + description.getClassName() + "/" + description.getMethodName()).getPath();
+
+      underTest = new PartFileWriter();
+      underTest.setOutputDirectoryPath(outputPath);
+
+      try {
+        File outDir = new File(outputPath);
+        FileUtils.forceMkdir(outDir);
+
+        blocksDir = new File(outputPath);
+        blocksDir.mkdirs();
+
+        populateBlocks();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    protected void populateBlocks()
+    {
+      long blockID = 1000;
+      for (int i = 0; i < FILE_CONTENTS.length; i++) {
+        int blockIndex = 1;
+        AbstractFileSplitter.FileMetadata fileMetadata = new AbstractFileSplitter.FileMetadata(outputPath);
+        fileMetadata.setRelativePath("file" + i);
+        int noOfBlocks = (int)((FILE_CONTENTS[i].length() / BLOCK_SIZE) + (((FILE_CONTENTS[i].length() % BLOCK_SIZE) == 0) ? 0 : 1));
+        long[] blockIds = new long[noOfBlocks];
+        for (int offset = 0; offset < FILE_CONTENTS[i].length(); offset += BLOCK_SIZE, blockID++) {
+          String blockContents;
+          if (offset + BLOCK_SIZE < FILE_CONTENTS[i].length()) {
+            blockContents = FILE_CONTENTS[i].substring(offset, offset + BLOCK_SIZE);
+          } else {
+            blockContents = FILE_CONTENTS[i].substring(offset);
+          }
+
+          AbstractBlockReader.ReaderRecord<Slice> readerRecord = new AbstractBlockReader.ReaderRecord<Slice>(blockID, new Slice(blockContents.getBytes()));
+          blockIds[blockIndex - 1] = blockID;
+          blockIndex++;
+          blockIdToExpectedContent.put(blockID, blockContents);
+          blockDataList.add(readerRecord);
+        }
+        fileMetadata.setBlockIds(blockIds);
+        fileMetadata.setNumberOfBlocks(noOfBlocks);
+        fileMetadatas.add(fileMetadata);
+      }
+    }
+
+    /* (non-Javadoc)
+     * @see org.junit.rules.TestWatcher#finished(org.junit.runner.Description)
+     */
+    @Override
+    protected void finished(Description description)
+    {
+      super.finished(description);
+
+      try {
+        FileUtils.deleteDirectory(new File(outputPath));
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  @Rule
+  public PartFileWriterTest.TestMeta testMeta = new PartFileWriterTest.TestMeta();
+
+  @Test
+  public void testBlockWriting() throws IOException
+  {
+    Attribute.AttributeMap attributes = new Attribute.AttributeMap.DefaultAttributeMap();
+    attributes.put(DAG.DAGContext.APPLICATION_ID, "PartitionWriterTest");
+    attributes.put(DAG.DAGContext.APPLICATION_PATH, testMeta.outputPath);
+    testMeta.context = mockOperatorContext(1, attributes);
+
+    testMeta.underTest.setup(testMeta.context);
+    int index = 0;
```

Comment:
It's confusing to have i, j, and index here. Renaming them will make this clearer.